CONTRIBUTING.md

```diff
@@ -325,6 +328,113 @@ if you change the history (rebase, squash, amend), and use git push --force to
 update the branch on your fork. The pull request points to that branch, not to
 specific commits.
 
+## Exercism Web Structure
+
+### Intro
+
+Exercism is built with the [Sinatra](https://github.com/sinatra/sinatra) web framework.
+If you haven't heard of it, definitely check it out. It's an awesome lightweight web framework for Ruby.
+
+Exercism seems to follow what some may call an MVP (Model-View-Presenter) architecture.
+Read more about that [here](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93presenter).
+
+### Core Directories
+
+#### API
+
+API contains routes which are used by the JavaScript frontend and the [CLI](https://github.com/exercism/cli).
+The routes here are similar to the one's in `app/` in that they're individual Sinatra apps that inherit from the Core API route (`api/v1/routes/core.rb`).
+If you add a route, it needs to be added to `api/v1/routes.rb` and `api/v1/v1.rb`.
+
+#### App (Routes, Presenters, Helpers and Views)
+
+App is the user facing side of Exercism.
+It handles logging in (and other functions), making calls out to the database to store or get information,
+structuring HTML (that can later be styled), etc.
+
+**Helpers** can be anything that *helps* presentation of data. For Exercism, these are things like:
+wrappers for Markdown and Syntax Highlighting parsers, session handling, etc.
+Helpers are defined in individual modules under `app/helpers/*`.
+The full list of helpers is in `app/helpers.rb`, and if you add a new helper, this file needs to be edited.
+The convention is `ClassName: 'filename'`.
+
+**Presenters** are for showing information that would be beneficial to users or the views but maybe not the best way to store data.
+For example, in Exercism, most times are stored in UTC (generally most servers store time this way) but... users might want to see times in their own timezone.
+Presenters can take relevant information (like showing comment notifications on the dashboard) and transform it to be more personal.
+This convention hasn't been strictly enforced so that description varies between presenters.
+Presenters are defined in individual classes/modules under `app/presenters/*`.
+The presenters are loaded into the app in the `app/presenters.rb` file, and if you add a presenter, it needs to be added to that file.
+The convention is `ClassName: 'filename'`.
+
+**Routes** in Sinatra are kind of like a combination of routes and controllers in Rails.
+Each route is like an individual Sinatra app that inherits from `Core` (found in `app/routes/core.rb`).
+In a route, you can specify endpoints (like `exercism.io/login`) and define how the app should respond (i.e. accept some credentials and log the user in or handle errors accordingly).
```

Comment:
_In a route file_

We need to make the distinction between a route (e.g. `GET /tracks/:id`) and a route file, which contains an app with multiple routes.